Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANS/SANSConvertToQ.py

```diff
@@ -0,0 +1,247 @@
+        wavelength_cutoff = convert_to_q.wavelength_cutoff
+        use_gravity = convert_to_q.use_gravity
+        gravity_extra_length = convert_to_q.gravity_extra_length
+
+        qxy_name = "Qxy"
+        qxy_options = {"InputWorkspace": data_workspace,
+                       "OutputWorkspace": EMPTY_NAME,
+                       "MaxQxy": max_q_xy,
+                       "DeltaQ": delta_q,
+                       "AccountForGravity": use_gravity,
+                       "RadiusCut": radius_cutoff,
+                       "WaveCut": wavelength_cutoff,
+                       "OutputParts": output_parts,
+                       "ExtraLength": gravity_extra_length}
+        if wavelength_adjustment_workspace:
+            qxy_options.update({"WavelengthAdj": wavelength_adjustment_workspace})
+        if pixel_adjustment_workspace:
+            qxy_options.update({"PixelAdj": pixel_adjustment_workspace})
+
+        qxy_alg = create_unmanaged_algorithm(qxy_name, **qxy_options)
+        qxy_alg.execute()
+
+        reduced_workspace = qxy_alg.getProperty("OutputWorkspace").value
+        reduced_workspace = self._replace_special_values(reduced_workspace)
+
+        # Get the partial workspaces
+        sum_of_counts_workspace, sum_of_norms_workspace = self._get_partial_output(output_parts, qxy_alg, do_clean=True)
+
+        return reduced_workspace, sum_of_counts_workspace, sum_of_norms_workspace
+
+    def _get_partial_output(self, output_parts, alg, do_clean=False):
+        if output_parts:
+            sum_of_counts_workspace = alg.getProperty("SumOfCounts").value
+            sum_of_norms_workspace = alg.getProperty("sumOfNormFactors").value
+            if do_clean:
+                sum_of_counts_workspace = self._replace_special_values(sum_of_counts_workspace)
+                sum_of_norms_workspace = self._replace_special_values(sum_of_norms_workspace)
+        else:
+            sum_of_counts_workspace = None
+            sum_of_norms_workspace = None
+        return sum_of_counts_workspace, sum_of_norms_workspace
+
+    def _set_partial_workspaces(self, sum_of_counts_workspace, sum_of_norms_workspace):
+        """
+        Sets the partial output, ie the sum of the counts workspace and the sum of the normalization workspace
+        @param sum_of_counts_workspace: the sum of the counts workspace
+        @param sum_of_norms_workspace: the sum of the normalization workspace
+        """
+        self.declareProperty(MatrixWorkspaceProperty("SumOfCounts", '',
+                                                     optional=PropertyMode.Optional, direction=Direction.Output),
+                             doc="The sum of the counts workspace.")
+        self.declareProperty(MatrixWorkspaceProperty("SumOfNormFactors", '',
+                                                     optional=PropertyMode.Optional, direction=Direction.Output),
+                             doc="The sum of the normalizations workspace.")
```

Comment:
I wonder if it would make sense to have these as optional output properties declared in `PyInit()`, and provide the extra output if user has specified them? Then the `OutputParts` property could be removed, and there wouldn't be need for hard-coded workspace name prefixes.